File: go/example/MtExampleServiceImpl.go
package example

import "fmt"

type MtExampleServiceImpl struct { 
}

func NewMtExampleServiceImpl() *MtExampleServiceImpl{
    return &MtExampleServiceImpl{}
}

func (p *MtExampleServiceImpl) SetUserProfile(oProfile *UserProfile) (r int32, err error){
    //TODO 
    fmt.Println("SetUserProfile called")
    return
}

func (p *MtExampleServiceImpl) GetUserProfile(userId int32) (r *UserProfile, err error) {
    //TODO
    fmt.Println("GetUserProfile called")
    return
}

File: php/MtExampleServiceImpl.php
<?php

include_once 'INC_COMM.php';
include_once 'MtExampleService.php';

class MtExampleServiceImpl implements MtExampleServiceIf{
    public function SetUserProfile(\UserProfile $oProfile){
        return 0;
    }
    public function GetUserProfile($userId){
        return new \UserProfile();
    }
}

?>
